cwltool/job.py

```diff
@@ -613,6 +617,19 @@ def add_writable_directory_volume(
     ) -> None:
         """Append a writable directory mapping to the runtime option list."""
 
+    def _preserve_environment_on_containers_warning(
+        self, varname: Optional[str] = None
+    ) -> None:
+        """When running in a container, issue a warning."""
+        if varname is None:
+            _logger.warning(
+                "You have specified `--preserve-entire-environment` while running a container"
```

Comment:
We should say why this results in a warning (could hamper reproducibility / portability)